ui/src/settings.vue

```diff
@@ -684,11 +793,17 @@ export default {
         },
         submitUser(e) {
             e.preventDefault();
-            this.userEdit.profile = JSON.parse(this.profile||"{}");
-            this.userEdit.scopes = JSON.parse(this.scopes||"{}");
+            if(!this.checkValidJson(this.userEditJSON)) {
+                this.$notify({type: "error", text: "Json formatting Error"});
+                return;
+            }
+            Object.keys(this.scopeModel).forEach(entry => {
+                this.userEdit.scopes[entry] = Object.keys(this.scopeModel[entry]).filter(val=>this.scopeModel[entry][val] == true);
+            });
             this.userEdit.ext.openids[0] = this.openids;
             this.$http.put(Vue.config.auth_api+"/user/"+this.userEdit.sub,this.userEdit).then(res=>{
                 this.$notify({type: "success", text: res.data.message});
+                this.showModal = false;
                 this.userEdit = null;
                 this.users = [];
```

Comment:
Can't we just update the specific `this.users` so you don't have to invalidate the whole array?